onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -534,10 +549,16 @@ class PlannerImpl {
       ORT_RETURN_IF_ERROR(status);
     }
     for (size_t i = 0; i != locations.size(); ++i) {
+      size_t max_pc = plan_.execution_plan.size();
       const std::vector<OrtMemoryInfo>& loc = locations[i];
       if (loc.empty()) continue;
       plan_.allocation_plan[i].alloc_kind = AllocKind::kAllocateStatically;
+      std::string node_arg_name;
+      ort_value_name_idx_map_.GetName(static_cast<int>(i), node_arg_name);
+      auto node_arg = graph_viewer_.GetNodeArg(node_arg_name);
+      plan_.allocation_plan[i].value_type = utils::GetMLDataType(*node_arg);
```

Comment:
What are the cost of the lookup going to be here in cases where the memory analysis is not being used -- do/should we have a more direct path for getting the type information than a string-based search?

Reply:
Yes, the cost of lookup name is O(n), since this is vector. I added a static flag to guard it: only doing it if we are doing profiling. I dont know a faster way to lookup the node information.